### DISCUSSION TITLE
No way to hide a field only on the create form but not on the update form ?

### DISCUSSION THREAD

**Discussion by florentpoujol** [laravel/nova-issues#3402] [Q&A] [answered]
- Laravel Version: 8.41.0
- Nova Version: 3.25.0
- PHP Version: 7.4.15
- Operating System and Version: Ubuntu 20.04
- Browser type and version: FF 88

### Description:

Fields have the `showOn*()` and `hideWhen*()`. methods to conditionally show/hide field from index/details/create/edit forms.

I want my field to be display everywhere except on the create form, but no fidling with the methods allowed me to hide a field only on one of the form, it is always hidden or show on both.

### Detailed steps to reproduce the issue on a fresh Nova installation:
Examle :
```php
 // hides the field on both forms
Text::make('test')->hideWhenCreating(),

 // hides the field on both forms
Text::make('test')
    ->hideWhenCreating()
    ->showOnUpdating(true)

// this show the field on both forms
Text::make('test')
    ->exceptOnForms()
    ->showOnUpdating(true)
```

Thanks for your time.

**Comment by florentpoujol** [comment]
So I guess this is not possible ?

Due to the name of the methods and the implementation that makes a clear distinction between create and update, I would definitely expect this to work that way ?

**Comment by crynobone** [comment, marked as answer]
> So I guess this is not possible ?

I moved this to Discussion because it seems you having a problem using these features instead of "it's not possible".

![CleanShot 2021-05-21 at 07 14 05](https://user-images.githubusercontent.com/172966/119060326-8c374500-ba04-11eb-9fa6-c2531451e606.gif)

I created a demo repo and seems all above working as expected. See https://github.com/nova-issues/issue-3402/commit/19dbc280c10ea3c23a4985dab1cba36d5d4f777c

**Reply by florentpoujol** [reply]
Hey crynobone

Thanks for your answer.

It just wasn't working for us, so after investigating the code, I found out we were overriding the `removeNonUpdateFields()` but calling inside `isShownOnCreation()`...

So all is good, thanks for your time !
